app/models/katello/subscription_status.rb

```diff
@@ -43,7 +43,13 @@ def to_global(_options = {})
 
     def to_status(options = {})
       return UNKNOWN unless host.subscription_facet.try(:uuid)
-      status_override = 'unsubscribed_hypervisor' if host.subscription_facet.hypervisor && !host.subscription_facet.candlepin_consumer.entitlements?
+      if host.subscription_facet.hypervisor
```

Comment:
This looks a bit complex now, so I'm questioning how optimal the approach is. The code (before you modified it) also made me wonder why we handle unsubscribed_hypervisor separately. And I guess it's because we need information from the subscription facet so we can't rely completely on the `Katello::Candlepin::Consumer` service class.

As far as this method is concerned, we ultimately want to 'not care' about unsubscribed_hypervisor if the org is in sca mode. So you should check SCA first, then based on that check for unsubscribed hypervisor. Here's how the code might be refactored:

To help us move some of this logic out, edit  subscription_facet.rb
```
def unsubscribed_hypervisor?
  !self.organization.simple_content_access && self.hypervisor && self.candlepin_consumer.entitlements?
end
```

Now we can use it here:
```
status_override = 'unsubscribed_hypervisor' if host.subscription_facet.unsubscribed_hypervisor?
```

Reply:
Updated to this, thank you for the pointer and help.